File: ITapply/ITapply.Models/SearchObjects/CandidateSkillSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ITapply.Models.SearchObjects
{
    public class CandidateSkillSearchObject : BaseSearchObject
    {
        public int? CandidateId { get; set; }
        public int? SkillId { get; set; }
        public int? MinLevel { get; set; }
        public int? MaxLevel { get; set; }
    }
} 

File: ITapply/ITapply.UI/itapply_mobile/lib/models/search_objects/candidate_skill_search_object.dart
import 'package:itapply_mobile/models/search_objects/base_search_object.dart';
import 'package:json_annotation/json_annotation.dart';

part 'candidate_skill_search_object.g.dart';

@JsonSerializable()
class CandidateSkillSearchObject extends BaseSearchObject {
  int? CandidateId;
  int? SkillId;
  int? MinLevel;
  int? MaxLevel;

  CandidateSkillSearchObject({
    this.CandidateId,
    this.SkillId,
    this.MinLevel,
    this.MaxLevel,
    super.Page,
    super.PageSize,
    super.IncludeTotalCount,
    super.RetrieveAll,
  });

  @override
  Map<String, dynamic> toJson() => _$CandidateSkillSearchObjectToJson(this);
}
